x/rewards/keeper/iprpc.go

```diff
@@ -2,12 +2,70 @@ package keeper
 
 import (
 	"fmt"
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/lavanet/lava/utils"
 	"github.com/lavanet/lava/x/rewards/types"
 )
 
+func (k Keeper) FundIprpc(ctx sdk.Context, creator string, duration uint64, fund sdk.Coins, spec string) error {
+	// verify spec exists and active
+	foundAndActive, _, _ := k.specKeeper.IsSpecFoundAndActive(ctx, spec)
+	if !foundAndActive {
+		return utils.LavaFormatWarning("spec not found or disabled", types.ErrFundIprpc)
+	}
+
+	// check fund consists of minimum amount of ulava (duration * min_iprpc_cost)
+	minIprpcFundCost := k.GetMinIprpcCost(ctx).Amount.MulRaw(int64(duration))
+	if fund.AmountOf(k.stakingKeeper.BondDenom(ctx)).LT(minIprpcFundCost) {
```

Comment:
we should compare one month fund to one month min cost not one month fund to duration times min cost

Reply:
also please add a unitest that would find this by setting cost = mincost *1.1 and duration 2

Reply:
will add the test to the unitests branch